File: javascript/funciones.js
function validarNumeros(a, b) {
    // Usamos una estructura condicional para comparar si "a" es igual a "b".
    if (a === b) {
        // Si son iguales, retornamos true.
        return true;
    } else {
        // Si no son iguales, retornamos false.
        return false;
    }
}

// Solicitamos al usuario que ingrese un número y lo convertimos a entero.
const a = parseInt(prompt("Introduzca un número"), 10);

// Solicitamos al usuario que ingrese otro número y lo convertimos a entero.
const b = parseInt(prompt("Introduzca otro número"), 10);

// Llamamos a la función "validarNumeros" pasando los dos números ingresados como argumentos y almacenamos el resultado en "resultado".
const resultado = validarNumeros(a, b);

// Usamos una estructura condicional para verificar el valor de "resultado".
if (resultado) {
    // Si "resultado" es true, imprimimos "Son iguales".
    console.log("Son iguales");
} else {
    // Si "resultado" es false, imprimimos "Son distintos".
    console.log("Son distintos");
}

File: rubi/funciones.rb
# Función que valida si dos números son iguales
def validar_numeros(a, b)
  # Compara si a es igual a b y devuelve true o false
  a == b
end

# Declaramos variables para almacenar los números
print "Introduzca un número: "
a = gets.chomp.to_i

print "Introduzca otro número: "
b = gets.chomp.to_i

# Llamamos a la función validar_numeros y almacenamos el resultado
resultado = validar_numeros(a, b)

# Evaluamos el resultado y mostramos el mensaje correspondiente
if resultado
  puts "Son iguales"
else
  puts "Son distintos"
end